core/src/main/java/com/mapzen/android/graphics/MapInitializer.java

```diff
@@ -90,15 +89,13 @@ private void loadMap(final MapView mapView, String sceneFile, final OnMapReadyCa
     final List<SceneUpdate> sceneUpdates = sceneUpdateManager.getUpdatesFor(apiKey, locale,
         mapStateManager.isTransitOverlayEnabled(), mapStateManager.isBikeOverlayEnabled(),
         mapStateManager.isPathOverlayEnabled());
-    getTangramView(mapView).getMapAsync(new com.mapzen.tangram.MapView.OnMapReadyCallback() {
-      @Override public void onMapReady(MapController mapController) {
-        mapController.setHttpHandler(mapzenMapHttpHandler.httpHandler());
-        MapzenManager mapzenManager = MapzenManager.instance(mapView.getContext());
-        callback.onMapReady(
-            new MapzenMap(mapView, mapController, new OverlayManager(mapView, mapController,
-                mapDataManager, mapStateManager), mapStateManager, new LabelPickHandler(mapView),
-                new MarkerManager(mapController), sceneUpdateManager, locale, mapzenManager));
+    MapController controller = mapView.getTangramMapView().getMap(
```

Comment:
The reason I set the `mapController`'s `SceneLoadListener` to `null` is because: 
1. After the initial scene loads, the SDK considers the map "ready" and doesn't want to invoke the `OnMapReadyCallback` for other calls to change the scene etc (looking at your comment below, I could also update this to check the sceneId)
2. The SDK provides type safety around scene updates (https://github.com/mapzen/android/blob/master/core/src/main/java/com/mapzen/android/graphics/SceneUpdateManager.java) and is tested to ensure the calls to sceneUpdateAsync don't produce errors. I know there has been talk about supporting external scenes but at the moment, the SDK doesn't have first class support for this case. If developers care about errors we expect them to access the `MapController` directly and set a `SceneLoadListener` on this object. There will be future discussion around this expansion of the SDK though.

Does this reasoning make sense, are there things I'm not considering properly?

On a related note, I was planning on setting a `SceneLoadListener` as part of the marker restoration work here though (https://github.com/mapzen/android/issues/348).